### PR TITLE
More NCO-related updates

### PR DESCRIPTION
**Pull request by AlexanderRichert-NOAA**
### Summary

This PR adds an NCO template to support operational installations, as well as a list of approved packages to check against. These can get added to develop later.

### Testing

Tested on Acorn (alogin01).

### Applications affected

n/a

### Systems affected

n/a

### Dependencies

none

### Issue(s) addressed

https://github.com/NOAA-EMC/WCOSS2-requests/issues/8

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [x] All dependency PRs/issues have been resolved and this PR can be merged.

### CODE REVIEW

File: configs/templates/nco/approved_list.txt [hunk 1]
```diff
@@ -0,0 +1,225 @@
+awscli
```
Comment (climbfuji) [inline configs/templates/nco/approved_list.txt:1]: Would it make sense to sort this alphabetically?

File: spack-ext/lib/jcsda-emc/spack-stack/stack/stack_env.py [hunk 2]
```diff
@@ -100,7 +100,7 @@ def __init__(self, **kwargs):
```
Comment (climbfuji) [inline spack-ext/lib/jcsda-emc/spack-stack/stack/stack_env.py]: This is probably from the other open PR #1294 that needs to be merged first?

File: configs/templates/nco/approved_list.txt (hunk 1 above)
Comment (AlexanderRichert-NOAA) [reply configs/templates/nco/approved_list.txt:1]: No, it's based on a spreadsheet from NCO and I'm just doing some grep magic on the concretization output, so it's fine as it is.

File: spack-ext/lib/jcsda-emc/spack-stack/stack/stack_env.py (hunk 2 above)
Comment (AlexanderRichert-NOAA) [reply spack-ext/lib/jcsda-emc/spack-stack/stack/stack_env.py]: Correct

**Review by climbfuji — APPROVED** [review]

**Comment by AlexanderRichert-NOAA** [comment]
Merging since this doesn't touch other configs...

**Pull request merged**